### ISSUE TITLE
What is BootstrapFinetune?

### ISSUE THREAD

**Issue by TyTodd** [stanfordnlp/dspy#8547]
Is `BootstrapFinetune` just a model distillation from a teacher model to a student model. So the student model learns to give responses closer to the teacher model? Does it also optimize the prompt? If so, could this be clearer in the docs? Also, are there any optimizers that fine-tune model weights using the metric as a loss function via GRPO?

**Comment by okhat** [comment]
1) dspy.BootstrapFinetune is an offline RL algorithm for DSPy programs. It's essentially filtered behavior cloning. It's not necessarily about distillation. Most of the time, it's used to get a model to teach *itself* how to do a task. It doesn't change the prompts, but it can compose with prompt optimizers, like in dspy.BetterTogether.

2) dspy.GRPO already exists indeed. This is an online RL algorithm, generalized to work for any DSPy program.

See dspy.GRPO tutorials: https://dspy.ai/tutorials/rl_papillon/

**Comment by 2010b9** [comment]
I was searching for more information about how `BootstrapFinetune` works and bumped into this issue. Is there some documentation that explains how it works? I’ve looked into the documentation, but haven’t found anything explaining this is more depth. In the case of `MIPROv2`, for example, there’s some [nice documentation about it](https://dspy.ai/api/optimizers/MIPROv2/#how-miprov2-works), but not for the `BootstrapFinetune` (at least I couldn’t find it).

Like @TyTodd I was assuming the `BootstrapFinetune` was based in knowledge distillation, as the terms “distillation”, “student” and “teacher” appear in the documentation.

Thanks in advance!
